File: src/main/resources/META-INF/resources/dojotags/Label.js
define([ "dojo/_base/declare", "dojo/_base/lang", "dojo/html", "dojo/dom-style", "./_Widget" ],
		function(declare, lang, html, domStyle, _Widget) {
			return declare("dojotags.Label", [ _Widget ], {
				initialize : function(args) {
					this.inherited(arguments);
					this.model.set("text", args.text || "");
					// when model's text changes, update the content of the dom
					// node
					// for this label
					this.model.watch("text", lang.hitch(this, function(name, oldVal, newVal) {
						html.set(this.domNode, newVal);
					}));
				},

				createDijit : function(node) {
					html.set(node, this.model.get("text"));
				},
				
				doUpdate : function(event, data) {
					this.model.set("text", data);
				}
			});
		});

File: src/main/java/com/github/dojotags/model/Label.java
package com.github.dojotags.model;

public class Label extends Widget {
	private static final long serialVersionUID = 1L;

	public String getText() {
		return (String) model.get("text");
	}

	public void setText(String text) {
		model.put("text", text);
	}
	
}
